### DISCUSSION TITLE
How can I find all groups that reside within a given OU ?

### DISCUSSION THREAD

**Discussion by manne65-hd** [DirectoryTree/LdapRecord#628] [Q&A] [answered]
Hello there,
I'm still new to LdapRecord and I have not yet found a way to get all groups that reside within a given OU _(and below that OU)_

Currently after setting up a connection to my LDAP (Active Directory) I can use this to get ALL groups _("starting" from my given BASE_DN)_

```php
 $groups = \LdapRecord\Models\ActiveDirectory\Group::get();
```
But I would like to find a way to get only the groups from a given OU, while still being able to make my connection with a "higher" BASE_DN

I guess this will be some "general" query, and I hope someone can give me a hint.

Looking forward to your answers,
Manfred

**Comment by stevebauman** [comment, marked as answer]
Hi @manne65-hd,

I apologize, this isn't clear enough in the documentation, I'll update this.

You have to set the base DN of your query on the `Group` model like so:

https://ldaprecord.com/docs/core/v3/searching#base-dn

> **Important**: I'm using [automatic base DN](https://ldaprecord.com/docs/core/v3/searching/#automatic-base-dn-substitution) substitution below which replaces `{base}` with your connections configured base DN.

```php
use LdapRecord\Models\ActiveDirectory\Group;

$groupsInsideAccounting = Group::in('ou=Accounting,{base}')->get();
```

Hope this helps! Let me know if you have further questions 👍

**Reply by manne65-hd** [reply]
Hi Steve,
wow that was a quick answer :thumbsup:

And it works like a charm!

Thank you very much

**Reply by stevebauman** [reply]
Glad it worked for you, always happy to help! 🙏

**Reply by HshahLaFarga** [reply]
Hi everyone,
I have OU with groups and another OU inside that one with another groups, but I only want to get the direct childs from the first one.
How can i get only the direct childs on that ou?

**Reply by stevebauman** [reply]
Hi @manne65-hd,

You may use the `listing()` method on the query builder to indicate you do not want the search to be recursive:

```php
use LdapRecord\Models\ActiveDirectory\Entry;

$results = Entry::listing()->in('ou=Accounting,{base}')->get();
```
